iOS/Flows/Home/_New/ConversationViewControllerDataSource.swift

```diff
@@ -0,0 +1,166 @@
+//
+//  ConversationViewControllerDataSource.swift
+//  ConversationViewControllerDataSource
+//
+//  Created by Martin Young on 9/15/21.
+//  Copyright © 2021 Benjamin Dodgson. All rights reserved.
+//
+
+import Foundation
+import StreamChat
+
+typealias ConversationCollectionSection = ConversationCollectionViewDataSource.SectionType
+typealias ConversationCollectionItem = ConversationCollectionViewDataSource.ItemType
+
+class ConversationCollectionViewDataSource: CollectionViewDataSource<ConversationCollectionViewDataSource.SectionType,
+                                            ConversationCollectionViewDataSource.ItemType> {
+
+    enum SectionType: Hashable {
+        case basic(ChannelId)
+    }
+
+    enum ItemType: Hashable {
+        case message(MessageId)
+    }
+
+    var handleDeleteMessage: ((ChatMessage) -> Void)?
+
+    private let contextMenuDelegate: ContextMenuInteractionDelegate
+
+    override init(collectionView: UICollectionView) {
+        self.contextMenuDelegate = ContextMenuInteractionDelegate(collectionView: collectionView)
```

Comment:
Annoyingly the delegate ContextMenuDelegate needs to be an NSObject so I made a proxy object that inherits from NSObject. Alternatively I could make all datasource objects NSObject.